### ISSUE TITLE
[CI] mapper-flattened tests fail on Windows from setting SetOnce twice

### ISSUE THREAD

**Issue by droberts195** [elastic/elasticsearch#47428]
Strangely this problem only occurs on Windows.  It's easy to reproduce _on Windows_ - simply run:

```
gradlew.bat :x-pack:plugin:mapper-flattened:test
```

The failure is:

```
  2> java.lang.IllegalStateException: failed to load plugin class [org.elasticsearch.xpack.core.XPackPlugin]

        Caused by:
        java.lang.reflect.InvocationTargetException

            Caused by:
            org.apache.lucene.util.SetOnce$AlreadySetException: The object cannot be set twice!

    java.lang.NullPointerException
```

Some examples of failing builds are:

* https://gradle-enterprise.elastic.co/s/2kqsuw23z767s
* https://gradle-enterprise.elastic.co/s/xgvg5sy4uwv5s
* https://gradle-enterprise.elastic.co/s/qjzljx6s6olow

**Comment by jimczi** [comment]
This looks similar to https://github.com/elastic/elasticsearch/pull/47252 but no idea why it is always reproducible on Windows

**Comment by henningandersen** [comment]
Fixed by #47252. The problem was not windows specific, rather a limited max-workers would reproduce this consistently. These packages have few tests only and if every test ran in its own JVM (due to test parallellization), the problem would not occur.
